### ISSUE TITLE
Can't set ELEMENTS_ROOT in QtCreator IDE when building out of source

### ISSUE THREAD

**Issue by redtide** [cycfi/elements#287]
In the examples' CMake files, when building out of the Elements source tree and `ELEMENTS_ROOT` is not set, an error is thrown.
This seems to cause difficulties for QtCreator IDE users, probably due the fact that files generation is skipped and then the configuration is not filled and disabled as consequence:
![immagine](https://user-images.githubusercontent.com/1171962/147940059-d0310b4b-8275-4973-abcd-246d67188af5.png)

I guess it's undesirable as a solution to replace the error with a warning, though moving the rest of the CMake settings inside an `else()` statement, producing an empty but working configuration, will allow to set the missing variable:
![immagine](https://user-images.githubusercontent.com/1171962/147940305-475a4bca-1035-4029-b459-e3226dafa9ba.png)

Basically a:
```cmake
if (NOT ELEMENTS_ROOT)
   message(WARNING "ELEMENTS_ROOT is not set")
else()
  # Config here
endif()
```
instead of a:
```cmake
if (NOT ELEMENTS_ROOT)
   message(FATAL_ERROR "ELEMENTS_ROOT is not set")
endif()
# Config here
```
